File: Jongryul/04/25192.py
import sys
input = sys.stdin.readline

n = int(input())
st = set()
cnt = 0
for _ in range(n):
	s = input().rstrip()
	if s == 'ENTER':
		st = set()
	elif s not in st:
		st.add(s)
		cnt += 1
print(cnt)

File: Season 1/15th_Competition/Jongryul/04/25192.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";
const input = fs
  .readFileSync(filePath)
  .toString()
  .trim()
  .split("\n")
  .map((l) => l.replace("\r", ""));
const n = Number(input[0]);
let cnt = 0;
const set = new Set();
let idx = 1;
for (let i = 0; i < n; i++) {
  const s = input[idx++];
  if (s === "ENTER") {
    set.clear();
  } else if (!set.has(s)) {
    set.add(s);
    cnt++;
  }
}
console.log(cnt);
